enacts/onset/maproom.py

```diff
@@ -622,73 +676,104 @@ def onset_tile(tz, tx, ty):
             dry_spell_length,
             dry_spell_search,
         )
+        if ("length" in map_choice) | ("total" in map_choice):
+            soil_moisture = calc.water_balance(
+                precip_tile, 5, 60, 0
+            )["soil_moisture"]
+            cess_dates = calc.seasonal_cess_date(
+                soil_moisture,
+                start_cess_day,
+                start_cess_month1,
+                search_days_cess,
+                water_balance_cess,
+                dry_spell_cess,
+            )
+            if "length" in map_choice:
+                if cess_dates["T"][0] < onset_dates["T"][0]:
+                    cess_dates = cess_dates.isel({"T": slice(1, None)})
+                    if cess_dates["T"].size != onset_dates["T"].size:
+                        onset_dates = onset_dates.isel({"T": slice(None, -2)})
+                seasonal_length = (
+                    (cess_dates["T"] + cess_dates["cess_delta"]).drop_indexes("T")
+                    - (onset_dates["T"] + onset_dates["onset_delta"]).drop_indexes("T")
+                ) #.astype("timedelta64[D]")
         if map_choice == "mean":
-            mymap = onset_dates.onset_delta.mean("T")
-            mymap_max = np.timedelta64(search_days, 'D')
+            map_data = onset_dates.onset_delta.mean("T")
+            map_max = np.timedelta64(search_days, 'D')
         if map_choice == "stddev":
-            mymap = onset_dates.onset_delta.dt.days.std(dim="T", skipna=True)
-            mymap_min = 0
```

Comment:
did you mean to remove this var def from the stddev if statement?

Reply:
yes. As of now, the min is 0 for all maps so it's defined before the series of if statements.